### ISSUE TITLE
Site policy duplicates content when using caseTransform

### ISSUE THREAD

**Issue by jmendeza** [craftercms/craftercms#7252]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [ ] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

Saving an existing asset will result in two different files in the repo when the file is matching a `caseTransform` rule. e.g.:
```
    <statement>
        <target-path-pattern>/(?!scripts/.*|templates/.*|static-assets/app/.*).*</target-path-pattern>
        <permitted>
            <path>
                <source-regex>(\.{2\,})|[^\w\-\./]</source-regex>
                <target-regex caseTransform="lowercase">-</target-regex>
            </path>
        </permitted>
    </statement>
```

### Steps to reproduce

Steps:

1. Create a new site from Editorial bp
2. From the sidebar, navigate to static-assets
3. Edit LICENSE.txt
4. Make any modification
5. Save
6. The UI now displays the asset with name in lower case
![Screenshot from 2024-09-05 09-41-07](https://github.com/user-attachments/assets/662a4158-e755-43f8-aa0c-b6f6c57223b7)
7. However, the repo contains both versions:
```
static-assets
├── css
├── fonts
├── images
├── js
├── license.txt
├── LICENSE.txt
└── README.txt
```

**Comment by jmendeza** [comment]
Actually this is not particular to `caseTransform` rule. Also regex transformation might create copies.
e.g.: manually add an asset with a space in its filename (`name contains spaces.txt`), then edit it in the UI: file is save as `name-contains-spaces.txt` and `name contains spaces.txt` remains locked

**Comment by phuongnq** [comment]
@sumerjabri @jmendeza What would be the expected behavior for this?
When writing content, the `@ValidateAction(type = Type.CREATE)` is applied and the policy will try to modify the path using the rule.
The backend does not separate a write if it is creating new or an edit to existing content.
Should these kinds of policies only apply to create new actions instead of edit actions?
We can modify the policy to make an exception for files such as LICENSE.txt; README.txt but shouldn't the developer not create files against their configured policy outside of Studio?

**Comment by sumerjabri** [comment]
@phuongnq please make it such that the policy engine only runs against `create`, not exiting documents.

**Comment by Euquimides** [comment]
Hi! Updated and rechecked this issue. Fix verified for v4.1.7 and v4.2.0. Thank you! Closing this ticket.
